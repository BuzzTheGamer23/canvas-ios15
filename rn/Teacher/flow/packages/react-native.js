// @flow
export type PanResponderGestureState = {
  stateID: number,
  moveX: number,
  moveY: number,
  x0: number,
  y0: number,
  dx: number,
  dy: number,
  vx: number,
  vy: number,
}
