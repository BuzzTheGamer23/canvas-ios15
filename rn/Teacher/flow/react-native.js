//
// Copyright (C) 2017-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow
/* eslint-disable */

// define basic style prop types
// the internal ones are the best we have
import {
  type ____ViewStyleProp_Internal,
  type ____TextStyleProp_Internal,
  type ____ImageStyleProp_Internal,
} from 'react-native/Libraries/StyleSheet/StyleSheetTypes'

declare type ViewStyleProp = ____ViewStyleProp_Internal
declare type TextStyleProp = ____TextStyleProp_Internal
declare type ImageStyleProp = ____ImageStyleProp_Internal
