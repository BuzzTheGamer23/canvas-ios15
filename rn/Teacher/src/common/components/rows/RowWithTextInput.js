//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow

import React, { Component } from 'react'
import {
  TextInput,
  StyleSheet,
  View,
} from 'react-native'
import Row, { type RowProps } from './Row'

type Props = RowProps & {
  defaultValue?: ?string,
  onChangeText?: (text: string, identifier: string) => void,
  inputWidth?: number, // only applies with title, default is 50
  inputHeight?: number, // only applies with title, default is 50
  placeholder?: ?string,
  keyboardType?: string,
  onFocus?: Function,
}

const ACCESSIBILITY_TRAITS = ['button']

export default class RowWithTextInput extends Component<any, Props, any> {

  render () {
    return this.props.title ? this._renderWithTitle() : this._renderWithoutTitle()
  }

  _renderWithoutTitle () {
    return <Row {...this.props} children={this._renderTextInput()} accessibilityTraits={ACCESSIBILITY_TRAITS} />
  }

  _renderWithTitle () {
    const accessory = (
      <View
        style={{
          height: this.props.inputHeight || 50,
          width: this.props.inputWidth || 50,
        }}
        children={this._renderTextInput({ textAlign: 'right' })}
      />
    )
    return <Row {...this.props} accessories={accessory} accessibilityTraits={ACCESSIBILITY_TRAITS} />
  }

  _renderTextInput (styles?: any) {
    return (
      <TextInput
        defaultValue={this.props.defaultValue}
        onChangeText={this._onChangeText}
        testID={this.props.identifier}
        style={[style.input, styles]}
        placeholder={this.props.placeholder}
        keyboardType={this.props.keyboardType}
        onFocus={this.props.onFocus}
      />
    )
  }

  _onChangeText = (text: string) => {
    if (this.props.onChangeText) {
      this.props.onChangeText(text, this.props.identifier)
    }
  }
}

const style = StyleSheet.create({
  input: {
    flex: 1,
  },
})
