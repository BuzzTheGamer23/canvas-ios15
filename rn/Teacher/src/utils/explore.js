/* @flow */

function hasProp (object: any, key: string, value: any): boolean {
  if (object instanceof Object) {
    const props = object.props
    if (props && props instanceof Object) {
      for (var prop in props) {
        if (prop === key && props[prop] === value) {
          return true
        }
      }
    }
  }

  return false
}

function selectByProp (object: any, propKey: string, value: any): ?any {
  if (!(typeof object === 'object')) {
    return null
  }

  if (hasProp(object, propKey, value)) {
    return object
  }

  let result = null

  for (var key in object) {
    if (object[key] instanceof Array) {
      object[key].forEach((e) => {
        result = selectByProp(e, propKey, value)
        if (result) {
          return result
        }
      })
    }

    result = selectByProp(object[key], propKey, value)
    if (result) {
      return result
    }
  }

  return result
}

class ComponentExplorer {
  component: any

  constructor (component: any) {
    this.component = component
  }

  selectByProp (propKey: string, value: any): ?any {
    return selectByProp(this.component, propKey, value)
  }

  selectByID (id: string): ?any {
    return this.selectByProp('testID', id)
  }
}

export default function explore (component: any): ComponentExplorer {
  return new ComponentExplorer(component)
}
