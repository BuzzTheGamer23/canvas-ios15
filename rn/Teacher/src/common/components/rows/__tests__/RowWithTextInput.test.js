//
// Copyright (C) 2016-present Instructure, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

/* @flow */

import React from 'react'
import 'react-native'
import renderer from 'react-test-renderer'

import RowWithTextInput from '../RowWithTextInput'
import explore from '../../../../../test/helpers/explore'

jest
  .mock('Button', () => 'Button')
  .mock('TouchableHighlight', () => 'TouchableHighlight')
  .mock('TouchableOpacity', () => 'TouchableOpacity')

describe('RowWithTextInput', () => {
  it('renders', () => {
    const tree = renderer.create(
      <RowWithTextInput />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('sends text changes', () => {
    const onChangeText = jest.fn()
    const tree = renderer.create(
      <RowWithTextInput
        onChangeText={onChangeText}
        identifier='test'
      />
    ).toJSON()
    const input: any = explore(tree).selectByID('test')
    input.props.onChangeText('changed')
    expect(onChangeText).toHaveBeenCalledWith('changed', 'test')
  })

  it('sets default value', () => {
    const tree = renderer.create(
      <RowWithTextInput
        identifier='test-default-value'
        defaultValue='test default value'
      />
    ).toJSON()
    const input: any = explore(tree).selectByID('test-default-value')
    expect(input.props.defaultValue).toEqual('test default value')
  })

  it('renders with a title', () => {
    expect(
      renderer.create(
        <RowWithTextInput title='Title' />
      ).toJSON()
    ).toMatchSnapshot()
  })
})
