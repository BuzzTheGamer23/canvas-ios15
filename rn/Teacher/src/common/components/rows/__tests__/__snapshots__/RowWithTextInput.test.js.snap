// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RowWithTextInput renders 1`] = `
<TouchableHighlight
  accessibilityLabel={undefined}
  accessibilityTraits={
    Array [
      "button",
    ]
  }
  onPress={[Function]}
  style={
    Array [
      Object {
        "height": 54,
      },
      undefined,
      undefined,
    ]
  }
  testID={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "flex": 1,
          "flexDirection": "row",
          "paddingBottom": 12.8,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 12.8,
        },
        Object {
          "backgroundColor": "white",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <TextInput
        defaultValue={undefined}
        keyboardType={undefined}
        onChangeText={[Function]}
        onFocus={undefined}
        placeholder={undefined}
        style={
          Array [
            Object {
              "flex": 1,
            },
            undefined,
          ]
        }
        testID={undefined}
      />
    </View>
  </View>
</TouchableHighlight>
`;

exports[`RowWithTextInput renders with a title 1`] = `
<TouchableHighlight
  accessibilityLabel={undefined}
  accessibilityTraits={
    Array [
      "button",
    ]
  }
  onPress={[Function]}
  style={
    Array [
      Object {
        "height": 54,
      },
      undefined,
      undefined,
    ]
  }
  testID={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "flex": 1,
          "flexDirection": "row",
          "paddingBottom": 12.8,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 12.8,
        },
        Object {
          "backgroundColor": "white",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        numberOfLines={0}
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "fontWeight": "600",
              },
              undefined,
            ],
            Object {
              "fontFamily": ".SFUIDisplay-semibold",
            },
          ]
        }
      >
        Title
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 0,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-end",
          }
        }
      >
        <View
          style={
            Object {
              "height": 50,
              "width": 50,
            }
          }
        >
          <TextInput
            defaultValue={undefined}
            keyboardType={undefined}
            onChangeText={[Function]}
            onFocus={undefined}
            placeholder={undefined}
            style={
              Array [
                Object {
                  "flex": 1,
                },
                Object {
                  "textAlign": "right",
                },
              ]
            }
            testID={undefined}
          />
        </View>
      </View>
    </View>
  </View>
</TouchableHighlight>
`;
